pkg/controller/job/job_controller.go

```diff
@@ -1705,6 +1715,9 @@ func isPodFailed(p *v1.Pod, job *batch.Job) bool {
 	if p.Status.Phase == v1.PodFailed {
 		return true
 	}
+	if feature.DefaultFeatureGate.Enabled(features.JobPodReplacementPolicy) && *job.Spec.PodReplacementPolicy == batch.Failed {
```

Comment:
@alculquicondor and @mimowo this was added to handle the case of removing finalizers only if pod is fully terminated (in cases of Failed)